src/elastic-san/azext_elastic_san/custom.py

```diff
@@ -9,6 +9,72 @@
 # pylint: disable=too-many-statements
 
 from knack.log import get_logger
-
+from .aaz.latest.elastic_san.volume_group import Create as _VolumeGroupCreate
+from .aaz.latest.elastic_san.volume_group import Update as _VolumeGroupUpdate
 
 logger = get_logger(__name__)
+
+
+class VolumeGroupCreate(_VolumeGroupCreate):
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg, AAZResourceIdArgFormat
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.identity.user_assigned_identity_id = AAZStrArg(
+            options=["user-assigned-identity"],
+            help="The ARM resource identifier of the User Assigned identity that will be used with this volume group. ",
+            fmt=AAZResourceIdArgFormat(template="/subscriptions/{subscription}/resourceGroups/{resource_group}"
+                                                "/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{}")
+        )
+        args_schema.identity.user_assigned_identities._registered = False
+        args_schema.encryption_properties.identity.user_assigned_identity._fmt = \
+            AAZResourceIdArgFormat(template="/subscriptions/{subscription}/resourceGroups/{resource_group}"
+                                            "/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{}")
+        return args_schema
+
+    def pre_operations(self):
+        args = self.ctx.args
+        if args.identity.user_assigned_identity_id:
+            uai_id = str(args.identity.user_assigned_identity_id)
+            args.identity["user_assigned_identities"] = {
+                uai_id: {}}
+            del args.identity.user_assigned_identity_id
+
+
+class VolumeGroupUpdate(_VolumeGroupUpdate):
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg, AAZResourceIdArgFormat
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.identity.user_assigned_identity_id = AAZStrArg(
+            options=["user-assigned-identity"],
+            help="The ARM resource identifier of the User Assigned identity that will be used with this volume group. ",
+            fmt=AAZResourceIdArgFormat(template="/subscriptions/{subscription}/resourceGroups/{resource_group}"
+                                                "/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{}")
+        )
```

Comment:
Is it nullable?

Reply:
Not according to service right now, might add in future releases.